torch/distributed/_sharding_spec/_internals.py

```diff
@@ -122,3 +122,31 @@ def check_tensor(shards_metadata, tensor_dims) -> None:
             f'Total volume of shards: {total_shard_volume} '
             f'does not match tensor volume: {tensor_volume}, in other words '
             f'all the individual shards do not cover the entire tensor')
+
```

Comment:
nit: seems what you are doing here is quite generic, not tied to tensor, probably reformat it to something more generic, sth like:

def get_chunk_size(N, chunks) --> Int 
"Splits N into chunks, returns the size of each chunk. 

def get_current_chunk_size(N, chunk_size, chunk_idx) --> Int

Alternatively something like

def split_to_chunks(N, chunks) --> List(int) might be more readable